applications/api-cat/src/test/java/no/acat/model/HarvestMetadataFactoryTest.java

```diff
@@ -0,0 +1,78 @@
+package no.acat.model;
+
+import no.dcat.shared.HarvestMetadata;
+import no.dcat.shared.testcategories.UnitTest;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+@RunWith(PowerMockRunner.class)
+@Category(UnitTest.class)
+public class HarvestMetadataFactoryTest {
+
+  Date harvestDate;
+  HarvestMetadata harvestMetadata;
+
+  @Before
+  public void setup() {
+    MockitoAnnotations.initMocks(this);
+    harvestDate = new DateTime(2018, 6, 20, 0, 0).toDate();
+    harvestMetadata = new HarvestMetadata();
+    harvestMetadata.setFirstHarvested(harvestDate);
+    harvestMetadata.setLastChanged(harvestDate);
+    harvestMetadata.setLastHarvested(harvestDate);
+  }
+
+  @PrepareForTest({HarvestMetadataFactory.class})
+  @Test
+  public void test_if_has_changed() throws Exception {
+
+    PowerMockito.mockStatic(HarvestMetadataFactory.class);
+    PowerMockito.doCallRealMethod()
+        .when(
+            HarvestMetadataFactory.class,
+            "recordHarvest",
+            Mockito.any(HarvestMetadata.class),
+            Mockito.any(Date.class),
+            Mockito.any(boolean.class));
+
+    HarvestMetadata metadata =
+        HarvestMetadataFactory.recordHarvest(harvestMetadata, harvestDate, true);
```

Comment:
Kan du forklare, hva er grunn, hvorfor trenger vi PowerMockito?
Etter jeg har lest hva PowerMockito er, det ser ut at det trenges, hvis man vil mock (bytte ut eller spy) statiske metoder.
Men her vi bare kaller en statisk metod, vi mocker jo ingenting. Vi må bare forme en inputobject, og assert output.

Reply:
PowerMokito er en populær rammeverk less : https://www.baeldung.com/intro-to-powermock
i dette tilfelle man trenger å test static ved å bruke PowerMockito.mockStatic. Det er best du leser gjennom koden: her tester vi recordHarvest og retur verdien. gjennom doCallRealMethod.
her vil jeg se actual og faktisk data. endret method navn til test_recordHarvest_if_metadata_has_changed.